src/login/components/LoginFormLedger/LoginFormLedger.js

```diff
@@ -66,36 +68,20 @@ export default class LoginFormLedger extends React.Component {
     );
   }
 
-  renderDeviceInfo = () => {
-    const { deviceInfo } = this.props;
+  renderStatus = () => {
+    const { publicKey, deviceError, deviceInfo, progress } = this.props;
 
-    if (deviceInfo) {
-      return <p>Found USB {deviceInfo.manufacturer} {deviceInfo.product}</p>;
+    if (deviceError !== 'No USB device found.') {
+      if (publicKey && deviceInfo && (progress !== FAILED)) {
```

Comment:
In conjunction with my previous comment, I don't think we should consider `publicKey` or `deviceInfo` anymore for conditionals now that `progress` is available.  We should just use `progress` alone to determine what messaging to show.

* If it's `LOADED`, then we know that `deviceInfo` & `publicKey` are available.
* If it's `FAILED`, then we know that `deviceError` is available.
* If it's neither of the above, then we know we're searching for USB device.

Reply:
Regarding the three comments above: I know that it looks a bit wonky at the moment. But the problem here is, that the progress changes it's state every second (when the inverval function refreshes the props). Therefore (if you output it to the console) you get a "LOADING", "LOADED", "LOADING", "LOADED", ... or a "LOADING", "FAILED", "LOADING", "FAILED", ... loop.

A simple progress === LOADED or progress === FAILED doesn't work at the moment, because of the LOADING steps between.

I thought about mapping LOADING and FAILED to handle them as false and only LOADED as true, but I failed with the result of a flickering deviceError message or a flickering connected to Ledger message.

I'll try to find a solution for that.

Reply:
Hmm, the `recentlyCompletedStrategy` is supposed to make it so we only see "LOADING" a single time, and once it progress to "LOADED" or "FAILED" it will never regress.  Is your experience seeing that before or after adding `recentlyCompletedStrategy`?

Reply:
For reference: https://github.com/neoverse/spunky#recently-completed-strategy

Reply:
I'm seeing this behavior on the current implementation.
Everytime the setInterval(this.props.poll, POLL_FREQUENCY);  executes, the progress returns 'LOADING' and some milliseconds later 'FAILED' or 'LOADED'.

I refactored the code to match your comments above and this is the last piece missing. Sadly the deviceError stays as it is even when the process changes to 'LOADED', otherwise I would have used it as a condition for success.

I could commit my current state of work to the branch, maybe I'm overseeing something trivial?

Reply:
@mhuggins I know you are not a fan of the state approach, but as long as recentlyCompletedStrategy returns the LOADING state, would the solution I provided in the last push feasible (working with componentDidUpdate and ignoring the LOADING states of progress)?